config/database.go

```diff
@@ -228,39 +230,53 @@ func (ds *DatabaseStore) persist(cfg *model.Config) error {
 		return errors.Wrap(err, "marshalled configuration failed length check")
 	}
 
-	tx, err := ds.db.Beginx()
-	if err != nil {
-		return errors.Wrap(err, "failed to begin transaction")
-	}
-	defer func() {
-		// Rollback after Commit just returns sql.ErrTxDone.
-		if err := tx.Rollback(); err != nil && err != sql.ErrTxDone {
-			mlog.Error("Failed to rollback configuration transaction", mlog.Err(err))
-		}
-	}()
-
+	sum := sha256.Sum256(b)
 	params := map[string]interface{}{
 		"id":        id,
 		"value":     value,
 		"create_at": createAt,
 		"key":       "ConfigurationId",
+		"sha":       hex.EncodeToString(sum[0:]),
 	}
 
 	// Skip the persist altogether if we're effectively writing the same configuration.
-	var oldValue []byte
-	row := ds.db.QueryRow("SELECT Value FROM Configurations WHERE Active")
-	if err := row.Scan(&oldValue); err != nil && err != sql.ErrNoRows {
+	var oldValue string
+	row := ds.db.QueryRow("SELECT SHA FROM Configurations WHERE Active")
+	if err = row.Scan(&oldValue); err != nil && err != sql.ErrNoRows {
 		return errors.Wrap(err, "failed to query active configuration")
 	}
-	if bytes.Equal(oldValue, b) {
+
+	oldSum, err := hex.DecodeString(oldValue)
+	if err != nil {
+		return errors.Wrap(err, "could not encode value")
+	}
+	if bytes.Equal(oldSum, sum[0:]) {
 		return nil
 	}
 
-	if _, err := tx.Exec("UPDATE Configurations SET Active = NULL WHERE Active"); err != nil {
-		return errors.Wrap(err, "failed to deactivate current configuration")
+	tx, err := ds.db.Beginx()
+	if err != nil {
+		return errors.Wrap(err, "failed to begin transaction")
+	}
+	defer func() {
+		// Rollback after Commit just returns sql.ErrTxDone.
+		if err = tx.Rollback(); err != nil && err != sql.ErrTxDone {
+			mlog.Error("Failed to rollback configuration transaction", mlog.Err(err))
+		}
+	}()
+
+	var oldId string
```

Comment:
A comment here would be helpful to clarify why we are taking this 2 query approach rather than one query.